packages/firebase_messaging/firebase_messaging/lib/src/messaging.dart

```diff
@@ -0,0 +1,368 @@
+  /// The Stream contains the [RemoteMessage].
+  ///
+  /// To handle messages whilst the app is in the background or terminated,
+  /// see [onBackgroundMessage].
+  static Stream<RemoteMessage> get onMessage {
+    Stream<RemoteMessage> onMessageStream =
+        FirebaseMessagingPlatform.onMessage.stream;
+
+    StreamController<RemoteMessage> streamController;
+    streamController = StreamController<RemoteMessage>.broadcast(onListen: () {
+      onMessageStream.pipe(streamController);
+    });
+
+    return streamController.stream;
+  }
+
+  /// Returns a [Stream] that is called when a user presses a notification message displayed
+  /// via FCM.
+  ///
+  /// A Stream event will be sent if the app has opened from a background state
+  /// (not terminated).
+  ///
+  /// If your app is opened via a notification whilst the app is terminated,
+  /// see [getInitialMessage].
+  static Stream<RemoteMessage> get onMessageOpenedApp {
+    Stream<RemoteMessage> onMessageOpenedAppStream =
+        FirebaseMessagingPlatform.onMessageOpenedApp.stream;
+
+    StreamController<RemoteMessage> streamController;
+    streamController = StreamController<RemoteMessage>.broadcast(onListen: () {
+      onMessageOpenedAppStream.pipe(streamController);
+    });
+
+    return streamController.stream;
+  }
+
+  /// Set a message handler function which is called when the app is in the
+  /// background or terminated.
+  ///
+  /// This provided handler must be a top-level function and cannot be
+  /// anonymous otherwise an [ArgumentError] will be thrown.
+  static void onBackgroundMessage(BackgroundMessageHandler handler) {
+    FirebaseMessagingPlatform.onBackgroundMessage = handler;
+  }
+
+  // ignore: public_member_api_docs
+  @Deprecated(
+      "Constructing Messaging is deprecated, use 'FirebaseMessaging.instance' instead")
+  factory FirebaseMessaging() {
+    return FirebaseMessaging.instance;
+  }
+
+  /// Returns whether messaging auto initialization is enabled or disabled for the device.
+  bool get isAutoInitEnabled {
+    return _delegate.isAutoInitEnabled;
+  }
+
+  /// If the application has been opened from a terminated state via a [RemoteMessage]
+  /// (containing a [Notification]), it will be returned, otherwise it will be `null`.
+  ///
+  /// Once the [RemoteMesage] has been consumed, it will be removed and further
+  /// calls to [getInitialMessage] will be `null`.
+  ///
+  /// This should be used to determine whether specific notification interaction
+  /// should open the app with a specific purpose (e.g. opening a chat message,
+  /// specific screen etc).
+  Future<RemoteMessage> getInitialMessage() {
+    return _delegate.getInitialMessage();
+  }
+
+  /// Removes access to an FCM token previously authorized with optional [senderId].
+  ///
+  /// Messages sent by the server to this token will fail.
+  Future<void> deleteToken() {
+    return _delegate.deleteToken();
+  }
+
+  /// On iOS, it is possible to get the users APNs token.
```

Comment:
```suggestion
  /// On iOS/MacOS, it is possible to get the users APNs token.
```

Otherwise, unclear what getAPNSToken() does on MacOS.